scalecube-configuration/src/main/java/io/scalecube/configuration/repository/ConfigurationRepository.java

```diff
@@ -41,7 +43,17 @@
    * @param doc the document to save.
    * @return Document instance by a given key.
    */
-  Mono<Document> save(String tenant, String repository, Document doc);
+  Mono<Document> createEntry(String tenant, String repository, Document doc);
+
+  /**
+   * Update a key in a tenant repository by key.
+   *
+   * @param tenant namespace of the repository.
+   * @param repository name to update key.
+   * @param doc the document to update.
+   * @return Document instance by a given key and version.
+   */
+  Mono<Document> updateEntry(String tenant, String repository, Document doc);
```

Comment:
Why do we need to change `Repository` interface for the sake of public API (service API) changes?